scripts/2015_nips_paper/run/score_ensemble.py

```diff
@@ -0,0 +1,236 @@
+
+    if isinstance(input_directories, str):
+        # add seed and task id directories
+        input_directories += '/%i/%i' % (seed, task_id)
+        input_directories = [input_directories]
+
+    else:
+        new_directories = []
+        for dir in input_directories:
+            dir += '/%i/%i' % (seed, task_id)
+            new_directories.append(dir)
+        input_directories = new_directories
+
+    validation_files = []
+    test_files = []
+    starttime = np.inf
+    for input_directory in input_directories:
+        print('Loading files from input directory:', input_directory)
+        validation_files_ = read_files(
+            os.path.join(input_directory,
+                         '.auto-sklearn/predictions_ensemble'),
+            n_jobs=n_jobs)
+        validation_files.extend(validation_files_)
+        test_files_ = read_files(
+            os.path.join(input_directory,
+                         '.auto-sklearn/predictions_test'),
+            n_jobs=n_jobs)
+        test_files.extend(test_files_)
+
+        assert len(validation_files_) > 0
+        assert len(validation_files_) == len(test_files_)
+
+        print('Loaded %d files!' % len(validation_files_))
+
+        # if not specified, we get all files.
+        seed_pattern = '*' if seed is None else str(seed)
+        glob_pattern = os.path.join(input_directory,
+                                    ".auto-sklearn",
+                                    "start_time_%s" % seed_pattern)
+        start_time_files = glob.glob(glob_pattern)
+
+        # find the earliest startime.
+        for start_time_file in start_time_files:
+            with open(start_time_file, "r") as fh:
+                starttime_candidate = float(fh.read())
+                if starttime_candidate < starttime:
+                    starttime = starttime_candidate
+
+        del validation_files_
+        del test_files_
+
+    validation_files.sort(key=lambda t: t[-1])
+
+    keys_to_test_files = {test_file[1]: test_file
+                          for test_file in test_files}
+    # Resort such that both files have the same order
+    test_files = [keys_to_test_files[validation_file[1]]
+                  for validation_file in validation_files]
+
+    assert [validation_file[1] for validation_file in validation_files] == [
+        test_file[1] for test_file in test_files]
+
+    losses = []
+    top_models_at_step = dict()
+
+    backend = create(input_directory, input_directory + "_output",
+                     delete_tmp_folder_after_terminate=False,
+                     delete_output_folder_after_terminate=True,
+                     shared_mode=True)
+    valid_labels = backend.load_targets_ensemble()
+    # validation_files[i]: validation prediction of i-th model
+    # validation_files[i][j]:
+    # validation files contain predictions of models
+    score = balanced_accuracy
+
+    for i in range(len(validation_files)):
+        loss = 1 - score(valid_labels, validation_files[i][0])
```

Comment:
Could you please adapt this to work both with scores and losses?

Reply:
(as well as below)

Reply:
Hi Matthias, do I understand correctly that you want the file to be able to deal with both the loss, which is actually balanced error rate (1 - balanced accuracy) in this case to be minimized, and score, which would be balanced accuracy to be maximized, by allowing users to specify it through the commandline argument?

Reply:
Hm, thinking about it again, maybe it is sufficient to point out that it only works with balanced accuracy.

Reply:
Okay, I will then write that on Readme.